libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/route/line/model/RouteSetValue.kt

```diff
@@ -18,12 +18,74 @@ import com.mapbox.maps.extension.style.expressions.generated.Expression
  */
 class RouteSetValue internal constructor(
     val primaryRouteSource: FeatureCollection,
-    val trafficLineExpression: Expression,
+    val trafficLineExpressionProvider: RouteLineExpressionProvider?,
     val routeLineExpression: Expression,
     val casingLineExpression: Expression,
-    val altRoute1TrafficExpression: Expression,
-    val altRoute2TrafficExpression: Expression,
+    val altRoute1TrafficExpression: RouteLineExpressionProvider?,
+    val altRoute2TrafficExpression: RouteLineExpressionProvider?,
     val alternativeRoute1Source: FeatureCollection,
     val alternativeRoute2Source: FeatureCollection,
     val waypointsSource: FeatureCollection
-)
+) {
+
+    /**
+     * @return a class with mutable values for replacing.
+     */
+    fun toMutableValue() = MutableRouteSetValue(
```

Comment:
This is only used for testing, let's not expose it.

Reply:
I did this based on the arch doc that defined how to mutate these values. This ticket seems like a good time to implement this since an expression producer method could be overridden by a developer. Adding it now means no semver issues in the future.